lib/Sema/TypeCheckCircularity.cpp

```diff
@@ -601,3 +607,43 @@ bool CircularityChecker::diagnoseInfiniteRecursion(CanType parentType,
 
   return true;
 }
+
+// Show a warning if the enum is inconstructible. The outcome of this method
+// is irrelevant.
+void CircularityChecker::diagnoseInconstructible(EnumDecl *E) {
+
+  auto containsType = [](TupleType *tuple, Type E) -> bool {
+    for (auto type: tuple->getElementTypes()) {
+      if (type->isEqual(E))
+        return true;
+    }
+    return false;
+  };
+  auto isInconstructible = [containsType, E]() -> bool {
```

Comment:
These are only used once so they can be pulled out as static functions.

Reply:
For the same reason, isn't it preferable to leave them in the function scope unless they are relatively long or complex? It's more readable and uniform this way.